### PR TITLE
Bugfix/userinfo project obs

### PR DESCRIPTION
**Pull request by borkarsaish65**

### CODE REVIEW

File: module/observations/helper.js [hunk 1]
```diff
@@ -217,7 +217,8 @@ module.exports = class ObservationsHelper {
     static createObservation(data,userId,solution,userRoleInformation="",userProfileInformation = {}) {
         return new Promise(async (resolve, reject) => {
             try {
-    
+                
+                let overWriteUserInfoIfReferenceFromProjectsIsFound = false;
```
Comment (aks30) [inline module/observations/helper.js]: @borkarsaish65 Don't use new flag just do it for if data.project is there

Comment (borkarsaish65) [reply module/observations/helper.js]: fixed

File: migrations/updateUserProfilesFromProjects/copyUserDataFromProjectsToObservations.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  const db = connection.db(dbName);
+
+  try {
+    console.log('Migration started...')
+    let observationIdsUpdated = [];
+    // Get all observations with referenceFrom "project"
+    const observationDocuments = await db
+      .collection("observations")
+      .find({ referenceFrom: "project" })
+      .toArray();
```
Comment (aks30) [inline migrations/updateUserProfilesFromProjects/copyUserDataFromProjectsToObservations.js:35]: @borkarsaish65 Project only the information which is needed, here only _id

File: migrations/updateUserProfilesFromProjects/copyUserDataFromProjectsToObservations.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    const totalChunkLength = chunkedObservations.length;
+    let iteration = 1;
+    for (const chunk of chunkedObservations) {
+      console.log(`processing chunk of ${iteration++}/${totalChunkLength}`)
+      const projectIds = chunk.map((obs) => obs.project._id);
+
+      // Fetch relevant projects
+      const projectRecords = await db
+        .collection("projects")
+        .find({ _id: { $in: projectIds } })
```
Comment (aks30) [inline migrations/updateUserProfilesFromProjects/copyUserDataFromProjectsToObservations.js:53]: @borkarsaish65 Project only the information which is needed

**Review by aks30 — CHANGES_REQUESTED** [review]
Reviewed on 26 Nov

File: migrations/updateUserProfilesFromProjects/copyUserDataFromProjectsToObservations.js (hunk 3 above)
Comment (borkarsaish65) [reply migrations/updateUserProfilesFromProjects/copyUserDataFromProjectsToObservations.js:53]: done

File: migrations/updateUserProfilesFromProjects/copyUserDataFromProjectsToObservations.js (hunk 2 above)
Comment (borkarsaish65) [reply migrations/updateUserProfilesFromProjects/copyUserDataFromProjectsToObservations.js:35]: done

**Pull request merged**